src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -562,7 +570,11 @@ bool SecurityManager::restore_discovered_participant_info(
     }
     else
     {
-        remove_discovered_participant_info(std::move(auth_ptr));
+                // AuthUniquePtr must be removed after unlock SecurityManager's mutex.
+                // This is to avoid a deadlock with TimedEvents.
+                DiscoveredParticipantInfo::AuthUniquePtr remote_participant_info = std::move(auth_ptr);
```

Comment:
Sorry. I messed this up on the web interface 🤦 

```suggestion
        // AuthUniquePtr must be removed after unlock SecurityManager's mutex.
        // This is to avoid a deadlock with TimedEvents.
        DiscoveredParticipantInfo::AuthUniquePtr remote_participant_info = std::move(auth_ptr);
```